subsys/shell/shell_wildcard.c

```diff
@@ -14,7 +14,7 @@ static void subcmd_get(const struct shell_cmd_entry *cmd,
 		       struct shell_static_entry *d_entry)
 {
 	__ASSERT_NO_MSG(entry != NULL);
-	__ASSERT_NO_MSG(st_entry != NULL);
```

Comment:
1. Seeing no `st_entry` in argument list, how could this have compiled before? Either this translation unit was never compiled with asserts or there is a global.
2. Does `cmd` deserve an assert too?
3. Does `*entry` deserve an assert too?

Reply:
@mped-oticon 
1. Argument name has changed and I've forgotten to update assert.
2. `cmd` does not need an assert, there is an `if` below.
3. Why it would need and assert This function is responsible to assign value for `*entry` variable.

Wildcard feature is not active by default hence it is not compiled. I need to update default shell configuration but in a separate PR.